dxp/resource_serving.md

```diff
@@ -0,0 +1,161 @@
+
+The algorithm to compute the ETag is a simple checksum computation that can be found in the [ETagUtil class](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/servlet/filters/etag/ETagUtil.java#L63).
+
+> Note that this filter just adds the `ETag` header to the response, but doesn't handle the rest of the caching control protocol that is necessary for HTTP.
+
+> Also note that, for dynamic resources, it may make more sense to avoid using this filter and compute `ETag`s with a better/safer/faster algorithm.
+
+### Cache Filter
+
+This filter is implemented in a single class but can be configured for friendly URLs, layouts and resources. We will describe the resources behavior
+only.
+
+Caches are maintained by `companyId`. Then, there's a portal property called `cache.filter.include.user.agent` that when set to `true` maintains caches per user agent, instead of a global one for all browsers. By default it is `true`.
+
+Basically this filter caches responses that are subject to caching (similar to what a proxy cache would do). Specifically it scans the `Cache-Control` header for `no-cache` but runs other additional tests to make sure that a request can be cached safely.
+
+When a cache hit occurs, the server returns the same response it sent last time (including headers, not just the content).
+
+### GZip Filter
+
+This filter compresses the response with the `gzip` algorithm. Its activation can be altered:
+
+1. Request parameter `compress`: when set to `false` filter is inhibited.
+2. The browser's `Accept-Encoding` header: unless `gzip` is accepted, the filter is disabled.
+
+### Language Filter
+
+This filter first determines the request's locale from a variety of sources: the `languageId` parameter, the `ThemeDisplay`, `doAsUserLanguageId` parameter, stored session value, user, cookies, ... See [the class' source code](https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/servlet/filters/language/LanguageFilter.java) for more detail.
+
+Once the locale is determined, it replaces each occurrence of `Liferay.Language.get(...)` by the translated value according to the deployed `Language.properties` files.
+
+> This filter is the reason why placing any `Liferay.Language.get(...)` call in your JavaScript source works synchronously without calling the server again or executing any code.
+
+> And of course, it is also the reason why dynamic language keys cannot be used in JavaScript.
+
+### Aggregate Filter
+
+This filter is used to bundle multiple JavaScript into a minified single resource (called bundle) that can be fetched from the server, as well as to minify single JavaScript or CSS files.
```

Comment:
"multiple JavaScript" → "multiple JavaScript files" (because "JavaScript" is un uncountable noun, you can't prefix it with "multiple").

Reply:
:+1: